.github/workflows/test-agent.yml

```diff
@@ -0,0 +1,143 @@
+# Copyright 2020 New Relic Corporation. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+
+name: test-pull-request
+
+#
+# Controls when the action will run. 
+#
+on:
+  #
+  # Triggers the workflow on push or pull request events but only for the main branch
+  #
+  push:
+    branches: 
+      - main
+      - 'dev'
+  pull_request:
+    branches: 
+      - main
+      - 'dev'
+
+jobs:
+  daemon-unit-tests:
+    runs-on: ubuntu-latest
```

Comment:
I'm torn on whether it would technically be safer to specify the version (like `ubuntu-22.04`) rather than `latest` just in case there's an issue with compatibility in a future release

Reply:
This begs the question if we need the ability to specify the target for these tests so we can try them on different OSes.